mirascope/core/base/tool.py

```diff
@@ -26,7 +27,7 @@ class ToolConfig(TypedDict, total=False):
     cache_control: _CacheControl
 
 
-class BaseTool(BaseModel):
+class BaseTool(Generic[_ToolSchema], BaseModel):
```

Comment:
hmm, I just noticed we don't actually employ this generic when subclassing this class (i.e. `OpenAITool` doesn't actually specify the generic).

Any reason we need this then? I'm also curious why we need the `tool_schema` method on `BaseTool` at all since it will never be implemented?

Reply:
This is to put the correct Tool type in `call_kwargs["tools"]`.
If we do not have this generics, we need to cast `tool_type.tool_schema()` to `_BaseToolT` in the following code.
`call_kwargs[“tools”] = [tool_type.tool_schema() for tool_type in tool_types]`

Reply:
Do you want to discontinue generics and put cast here?

Reply:
Ah I see, we should keep this, but we should update all of our tool definitions to use the generic then so it's clear?

for example, right now we have `class AnthropicTool(BaseTool): ...` but we should actually have `class AnthropicTool(BaseTool[ToolParam]): ...` right?

Reply:
> for example, right now we have class AnthropicTool(BaseTool): ... but we should actually have class AnthropicTool(BaseTool[ToolParam]): ... right?

It's good for type-checking aspect.
I applied the types